crates/egui/src/containers/scroll_area.rs

```diff
@@ -493,6 +493,7 @@ impl ScrollArea {
         } = self;
 
         let ctx = ui.ctx().clone();
+        let scrolling_enabled = scrolling_enabled && ui.is_enabled();
```

Comment:
If `disabled` means that you cannot change widget properties (like entering text in text fields), then scroll area should not respect this setting, because it does not have properties that reflects application data. Imagine a read-only table widget which you cannot scroll because is is non-editable. I think, that is not what you want.

Reply:
In mainstream UI frameworks like Qt and MFC, disabling a parent widget results in all child widgets becoming non-interactive. This includes the scroll bars of the child widgets.

Instead of relying on the disabled state of the scroll area, a read-only table should be implemented as an inherent feature of the table itself. This approach is more direct and explicit, and it avoids potential issues that could arise from disabling the scroll area.